src/platforms/esp32/main/spi_driver.c

```diff
@@ -47,18 +48,76 @@
 #include "esp32_sys.h"
 #include "sys.h"
 
-static void spidriver_consume_mailbox(Context *ctx);
+#define TAG "spi_driver"
 
-static uint32_t spidriver_transfer_at(Context *ctx, uint64_t address, int data_len, uint32_t data, bool *ok);
+struct SPIDevice
+{
+    struct ListHead list_head;
+    term device_name;
+    spi_device_handle_t handle;
+};
+
+struct SPIData
+{
+    spi_host_device_t host_device;
+    struct ListHead devices;
```

Comment:
when we are not sure about members size, let's prefer moving before members that we are sure that are multiple of word size, so we always avoid some padding bytes. so let's swap `devices` and `host_device`.